cpp/ql/lib/change-notes/2024-11-18-throwing-functions.md

```diff
@@ -0,0 +1,4 @@
+---
+category: minorAnalysis
+---
+* Removed NonThrowing.qll. Throwing meta-data now part of Throwing.qll. Updated models and IR to use the new Throwing library and predicates.
```

Comment:
A couple of suggestions:
- There's a category for deprecating stuff (see https://github.com/github/codeql/blob/main/docs/change-notes.md) that seems appropriate here.
- Users probably don't care that we updated the models since it doesn't have any observable behavior (yet). So I'd say there's no need to include this in a change note.

```suggestion
category: deprecated
---
* The `NonThrowing` class (`semmle.code.cpp.models.interfaces.NonThrowing`) has been deprecated. Please use the `NonThrowing` class from `semmle.code.cpp.models.interfaces.Throwing` instead.
```